app/assets/javascripts/pageflow/editor/api/file_types.js

```diff
@@ -29,26 +29,42 @@ pageflow.FileTypes = pageflow.Object.extend({
   },
 
   setup: function(serverSideConfigs) {
+    var clientSideConfigs = this.clientSideConfigs;
     this._setup = true;
 
-    this.fileTypes = _.map(serverSideConfigs, function(serverSideConfig) {
-      var clientSideConfig = this.clientSideConfigs[serverSideConfig.collectionName];
-
-      if (!clientSideConfig) {
-        throw 'Missing client side config for file type "' + serverSideConfig.collectionName + '"';
+    this.fileTypes = new pageflow.
+      FileTypesCollection(_.map(serverSideConfigs, function(serverSideConfig) {
+        var clientSideConfig = clientSideConfigs[serverSideConfig.collectionName];
+
+        if (!clientSideConfig) {
+          throw 'Missing client side config for file type "' +
+            serverSideConfig.collectionName + '"';
+        }
+
+        _(this.clientSideConfigModifications[serverSideConfig.collectionName])
+          .each(function(modification) {
+            this.lintModification(modification, serverSideConfig.collectionName);
+            this.applyModification(clientSideConfig, modification);
+          }, this);
+
+        return new pageflow.FileType(_.extend({}, serverSideConfig, clientSideConfig));
+      }, this));
+
+    _.map(this.fileTypes.fileTypes, function(fileType) {
```

Comment:
Why `map`? `fileTypes` is enumerable itself. No need to reach into its internal state.